android/engine/src/main/java/org/smartregister/fhircore/engine/ui/pin/PinLoginScreen.kt

```diff
@@ -0,0 +1,271 @@
+import org.smartregister.fhircore.engine.util.annotation.ExcludeFromJacocoGeneratedReport
+
+const val PIN_TOOLBAR_MENU = "toolbarMenuTag"
+const val PIN_TOOLBAR_MENU_BUTTON = "toolbarMenuButtonTag"
+const val PIN_TOOLBAR_TITLE = "toolbarTitle"
+const val PIN_TOOLBAR_MENU_ICON = "toolbarIcon"
+const val PIN_TOOLBAR_MENU_LOGIN = "toolbarMenuLogin"
+const val PIN_FORGOT_PIN = "forgotPin"
+const val PIN_FORGOT_DIALOG = "forgotPinDialog"
+const val PIN_TOOLBAR_MENU_SETTINGS = "toolbarMenuSettings"
+const val PIN_SET_PIN_CONFIRM_BUTTON = "setPinConfirmButton"
+
+@Composable
+fun PinLoginScreen(viewModel: PinViewModel) {
+
+  val showError by viewModel.showError.observeAsState(initial = false)
+
+  PinLoginPage(
+    onPinChanged = viewModel::onPinChanged,
+    showError = showError,
+    enterUserPinMessage = viewModel.enterUserLoginMessage,
+    onMenuLoginClicked = { viewModel.onMenuLoginClicked() },
+    forgotPin = viewModel::forgotPin,
+    appName = viewModel.appName
+  )
+}
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+fun PinLoginPage(
+  modifier: Modifier = Modifier,
+  onPinChanged: (String) -> Unit,
+  showError: Boolean = false,
+  onMenuLoginClicked: () -> Unit,
+  enterUserPinMessage: String = "",
+  forgotPin: () -> Unit,
+  appName: String = ""
+) {
+
+  var showMenu by remember { mutableStateOf(false) }
+  var showForgotPinDialog by remember { mutableStateOf(false) }
+
+  Surface(color = colorResource(id = R.color.white_slightly_opaque)) {
+    TopAppBar(
+      title = { Text(text = "", Modifier.testTag(PIN_TOOLBAR_TITLE)) },
+      navigationIcon = {
+        IconButton(onClick = {}) {
+          Icon(
+            Icons.Filled.ArrowBack,
+            contentDescription = "Back arrow",
+            modifier = Modifier.size(0.dp).testTag(PIN_TOOLBAR_MENU_ICON)
+          )
+        }
+      },
+      actions = {
+        IconButton(
+          onClick = { showMenu = !showMenu },
+          modifier = Modifier.testTag(PIN_TOOLBAR_MENU_BUTTON)
+        ) { Icon(imageVector = Icons.Outlined.MoreVert, contentDescription = null) }
+        DropdownMenu(
+          expanded = showMenu,
+          onDismissRequest = { showMenu = false },
+          Modifier.testTag(PIN_TOOLBAR_MENU)
+        ) {
+          DropdownMenuItem(
+            onClick = {
+              showMenu = false
+              onMenuLoginClicked()
+            },
+            modifier = Modifier.testTag(PIN_TOOLBAR_MENU_LOGIN)
+          ) { Text(text = stringResource(id = R.string.pin_menu_login)) }
+        }
+      }
+    )
+
+    if (showForgotPinDialog) {
+      ForgotPinDialog(forgotPin = forgotPin, onDismissDialog = { showForgotPinDialog = false })
+    }
+
+    Column(
+      modifier =
+        Modifier.fillMaxSize()
+          .padding(horizontal = 16.dp, vertical = 70.dp)
+          .wrapContentWidth(Alignment.CenterHorizontally)
+    ) {
+      val appLogo =
+        when (appName) {
+          "g6pd" -> R.drawable.ic_logo_g6pd
```

Comment:
this should be loaded from a config, is this code moved from somewhere else? maybe @shoaibmushtaq25's refactor didn't address this part of g6pd specific stuff?

Reply:
yeah, this appLogo logic is copied from LoginScreen